Clockwise/ConfigurationExtensions.cs

```diff
@@ -50,6 +51,62 @@ public static class ConfigurationExtensions
             return configuration;
         }
 
+        public static Configuration UseCircuitbreakerFor<TChannel, TCircuitBreaker>(this Configuration configuration) where TCircuitBreaker : CircuitBreaker
+        {
+            configuration.Container.AfterCreating<ICommandReceiver<TChannel>>(receiver =>
+            {
+                TCircuitBreaker cb;
+                try
+                {
+                    cb = configuration.Container.Resolve<TCircuitBreaker>();
+                    
+                }
+                catch (Exception e)
+                {
+                    throw new ConfigurationException($"Failure during creation of circuit breaker {typeof(TCircuitBreaker).Name}",e);
+                }
+
+                async Task<ICommandDeliveryResult> ReceivingMiddleware(HandleCommand<TChannel> handlerDelegate, TimeSpan? timeout, Func<HandleCommand<TChannel>, TimeSpan?, Task<ICommandDeliveryResult>> next)
+                {
+                    return await next(async delivery =>
+                    {
+                        var result = await handlerDelegate(delivery);
+                        return result;
+                    }, timeout);
+                }
+
+                IDisposable SubscribingMiddleware(HandleCommand<TChannel> handle, Func<HandleCommand<TChannel>, IDisposable> subscribe)
+                {
+                    return subscribe(async delivery =>
+                    {
+                        {
+                            var stateDescriptor = await cb.GetLastStateAsync();
+                            if ( stateDescriptor.State == CircuitBreakerState.Open) return delivery.Retry(stateDescriptor.TimeToLive);
+
+                            var result1 = await handle(delivery);
+                            switch (result1)
+                            {
+                                case CommandDelivery.PauseDeliveryResult<TChannel> pause:
+                                    await cb.SignalFailure(pause.PausePeriod);
+                                    break;
+                                default:
+                                    await cb.SignalSuccess();
+                                    break;
+                            }
+
+                            return result1;
+                        }
+                    });
+                }
+
+                var instrumented = receiver.UseMiddleware(
+                    receive: ReceivingMiddleware,
+                    subscribe: SubscribingMiddleware);
```

Comment:
These method names will be more readable if they're verbs instead of nouns, e.g. `Receive` and `Subscribe`.